### ISSUE TITLE
Ability to set path to ESLint

### ISSUE THREAD

**Issue by isiahmeadows** [AtomLinter/linter-eslint#1051]
## Issue Type

Feature Request

## Issue Description

While working on ESLint itself, I found it was not possible to use this plugin without `npm link && npm link eslint` (circular symlink, which is bad). Instead, I'd like to be able to specify the path to ESLint directly, so I can still use this while developing on it.

The symptom I found that led me here is that it was complaining about a [missing plugin with an unhelpful message](https://github.com/AtomLinter/linter-eslint/issues/883).

(Note: this *used* to provide this functionality, but had since dropped it.)

## Bug Checklist
- Restart Atom
- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Linter Eslint: Debug output here
```

**Comment by Arcanemagus** [comment]
There already is the ability to specify a hardcoded path to `eslint`... as long as it ends with "`eslint`" that is.

https://github.com/AtomLinter/linter-eslint/blob/7f791653480ce30c1ffe9ab99c6c84e34fe7e7fe/src/worker-helpers.js#L67-L70

If you specify an absolute path using the [Advanced Local Node Modules](https://github.com/AtomLinter/linter-eslint/blob/7f791653480ce30c1ffe9ab99c6c84e34fe7e7fe/package.json#L46-L51) setting then the path you give will be used directly, the only modification being `eslint` appended on the end.

> The symptom I found that led me here is that it was complaining about a missing plugin with an unhelpful message.

The error message is coming straight from ESLint, the issue you linked is about replacing that with instructions on how to fix it 😉.

> (Note: this used to provide this functionality, but had since dropped it.)

I'm not sure what you mean here?

**Comment by isiahmeadows** [comment]
So, should I specify `advancedLocalNodeModules: '..'`?

> The error message is coming straight from ESLint, the issue you linked is about replacing that with instructions on how to fix it 😉.

I'm aware. I was linking a little too tangentially. Sorry about that.

> I'm not sure what you mean here?

You used to, in previous versions of this plugin, request the path to `eslint/lib/cli-engine.js` rather than just `eslint`.

**Comment by Arcanemagus** [comment]
> You used to, in previous versions of this plugin, request the path to `eslint/lib/cli-engine.js` rather than just `eslint`.

Ah! That was quite fragile and hacky, it was replaced with a less fragile but still somewhat hacy method used currently in https://github.com/AtomLinter/linter-eslint/pull/873. 😉
